### PR TITLE
hello-rails-react:Hello-Rails-React-Final-PR

### PR DESCRIPTION
**Pull request by Kumilachew-g**
**In this PR, I have done the following tasks.**
- Created a new Rails app called 'hello-rails-react'.
     - Use the commend **rails new -j webpack my_new_rails_app** as described in the previous lesson's tutorial.
- Verifyied that the **jsbundling-rails gem** is added to your Gemfile.
- Initialized your project with Git.
- Installed needed node dependencies as described in the previous lesson's tutorial.
- Created a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- Updated webpack.config.js to include the babel-loader as described in the previous lesson's tutorial.
- Checked that my project has a Postgres database set up.
- Created a table for storing your messages and creating 4 different greetings.
- Created an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- Created a static view that will be the root of your app.
- Created the App component with react-router and render it in your static view.
- Created the Greeting component that will display a greeting. Set it up as a route in your App component.
- Created a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- Displayed the random greeting in your Greeting component.

### CODE REVIEW

File: app/javascript/packs/application.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+//
+// To reference this file, add <%= javascript_pack_tag 'application' %> to the appropriate
+// layout file, like app/views/layouts/application.html.erb
+
+// Uncomment to copy all static images under ../images to the output folder and reference
+// them with the image_pack_tag helper in views (e.g <%= image_pack_tag 'rails.png' %>)
+// or the `imagePath` JavaScript helper below.
+//
+// const images = require.context('../images', true)
+// const imagePath = (name) => images(name, true)
```
Comment (mukhammadrizooff) [inline app/javascript/packs/application.js]: - [ ] [REQUIRED] Please delete `unnecessary` comments and do not use `comment` like `/* eslint ... */`, if you have problem with `eslint` try to solve it yourself 🙏🏼

File: config/routes.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Rails.application.routes.draw do
+  get 'static_pages/index'
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+
+  # Defines the root path route ("/")
+  root "static_pages#index"
+
```
Comment (mukhammadrizooff) [inline config/routes.rb]: - [ ] [REQUIRED] Good job so far! A controller with one action should be created and a view for the `root` **route** is created like this  => `root#index`. It should include only `<div id="root">` The rest should be done with React and Redux. Please implement my suggestions 🙏🏼

File: package.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    "eslint-plugin-import": "^2.24.1",
+    "eslint-plugin-jsx-a11y": "^6.6.1",
+    "eslint-plugin-react": "^7.31.11",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "stylelint": "^13.13.1",
+    "stylelint-config-standard": "^21.0.0",
+    "stylelint-csstree-validator": "^1.9.0",
+    "stylelint-scss": "^3.21.0",
+    "webpack-dev-server": "^4.11.1"
+  }
```
Comment (mukhammadrizooff) [inline package.json]: - [ ] [REQUIRED] Good job so far! The package.json file should be needed dependencies were added:
* `react-dom` ✔️ 
* `babel-loader` ❌ 
But I can't find `babel-loader` dependence in your `package.json` file, please implement it 🙏🏼

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @Kumilachew-g, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Required Changes ♻️
- [ ] [REQUIRED] Good job so far! Please extract the files from the `packs` folder because everything should exist in the `app/javascript/application.js` file 🙏🏼 

image

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 3 above)
Comment (Kumilachew-g) [reply package.json]: Thanks @mukhammadrizooff, I have implemented

File: config/routes.rb (hunk 2 above)
Comment (Kumilachew-g) [reply config/routes.rb]: Thanks @mukhammadrizooff, Just fixed

File: app/javascript/packs/application.js (hunk 1 above)
Comment (Kumilachew-g) [reply app/javascript/packs/application.js]: Thanks @mukhammadrizooff, Removed

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Kumilachew-g , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**